x/genutil/client/cli/gentx.go

```diff
@@ -33,6 +33,7 @@ import (
 func GenTxCmd(mbm module.BasicManager, txEncCfg client.TxEncodingConfig, genBalIterator types.GenesisBalancesIterator, defaultNodeHome string) *cobra.Command {
 	ipDefault, _ := server.ExternalIP()
 	fsCreateValidator, defaultsDesc := cli.CreateValidatorMsgFlagSet(ipDefault)
+	fsCreateValidator.String(cli.FlagServerHome, "", "Node's configuration (\"home\") directory.")
```

Comment:
Is this the correct approach? `--home` is the directory that contains all configuration, data, and keybase file material. If anything, `gentx` should take a `--client-home` arg.

Reply:
I think you're right actually - it makes more sense this way. I have a question too: apart from the keyring, is there anything else that the client would read from client's home? What happens to client's config file?

Reply:
The problem with this PR is that it's half-baked. In other words, if we're doing this for the `gentx` command, then we should really be doing it for _all_ commands that use the Keyring.

I'm not saying we shouldn't merge this, but just keep this in mind.

Reply:
> we should really be doing it for all commands that use the Keyring.

I completely agree

Reply:
> The problem with this PR is that it's half-baked. In other words, if we're doing this for the `gentx` command, then we should really be doing it for _all_ commands that use the Keyring.

Agreed.  I can see about doing that more systematically.

Reply:
The latest push implements `--client-home` wherever `--keyring-backend` appeared.  That should be systematic enough.